case/src/main/kotlin/com/ritense/case_/domain/tab/CaseWidgetTabWidget.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2015-2024 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.case_.domain.tab
+
+import jakarta.persistence.Column
+import jakarta.persistence.DiscriminatorColumn
+import jakarta.persistence.DiscriminatorType
+import jakarta.persistence.Entity
+import jakarta.persistence.Id
+import jakarta.persistence.Inheritance
+import jakarta.persistence.InheritanceType
+import jakarta.persistence.Table
+
+@Entity
+@Table(name = "case_widget_tab_widget")
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+@DiscriminatorColumn(
+    name = "case_widget_type",
+    discriminatorType = DiscriminatorType.STRING
+)
+data class CaseWidgetTabWidget(
```

Comment:
How can we use the discriminator type when this is a (closed) data class? I don't see any test that proves that the inheritance is working.

Reply:
Fixed it.